packages/api/amplify_api/lib/src/decorators/web_socket_auth_utils.dart

```diff
@@ -109,7 +113,10 @@ Future<Map<String, String>> _generateAuthorizationHeaders(
   final maybeConnect = isConnectionInit ? '/connect' : '';
   final canonicalHttpRequest = AWSStreamedHttpRequest.post(
     Uri.parse('${config.endpoint}$maybeConnect'),
-    headers: _requiredHeaders,
+    headers: {
+      ..._requiredHeaders,
+      ...(customHeaders ?? {}),
```

Comment:
I would switch the order so that custom headers cannot overwrite required headers

```suggestion
      ...?customHeaders,
      ..._requiredHeaders,
```